packages/expo-notifications/src/__tests__/Notifications-test.ts

```diff
@@ -108,6 +132,55 @@ it(`verifies daily trigger handling with channelId`, async () => {
   );
 });
 
+it(`verifies weekly trigger handling with channelId`, async () => {
+  const trigger = {
+    weekday: 1,
+    hour: 12,
+    minute: 30,
+    channelId: 'test-channel-id',
+    repeats: true as boolean | undefined,
+  };
+  const input = {
+    ...notificationTriggerInputTest,
+    trigger,
+  };
+  await scheduleNotificationAsync(input);
+  delete trigger.repeats;
+  expect(NotificationScheduler.scheduleNotificationAsync).toHaveBeenLastCalledWith(
+    input.identifier,
+    input.content,
+    {
+      type: 'weekly',
+      ...input.trigger,
+    }
+  );
+});
+
+it(`verifies yearly trigger handling with channelId`, async () => {
+  const trigger = {
+    day: 1,
+    month: 6,
```

Comment:
what do you think of adding some tests to verify these inputs result in the expected dates? The iOS `month + 1` situation makes me want to ensure that we're getting the expected values in the future (when others change this code or add to it)

Reply:
I haven't dug into this, but I think these tests in particular don't run on device, or do they? Because I can run them without any si(e)mulator or device running… so in that case they won't reach the native iOS code with the `month + 1` sentence. Maybe I should add some in the `apps/test-suite/tests/NewNotifications.js` file?

Reply:
Yes, sorry for the confusion! These tests run in node. I should've specified adding to `apps/test-suite/tests/NewNotifications.js`

Reply:
This test here will be testing for this type of cases https://github.com/expo/expo/pull/11898/files#diff-81f5105968868ab9c836c95a9eba24f7843bf2a5f82dd9f48d75d3eaa04a56c0R1295-R1296

In Android it will expect the values will be the same, but in iOS it will expect all of them the same except for month which needs to be +1. I don't like too much to expect different things from JS (you send month, and you need to expect month+1) but I don't think we have a choice for the moment. The yearly (and daily/weekly) are a facade to Calendar trigger, which is iOS only, and it is already expecting month 1-12 as property, since it passes the data directly to iOS type… so the `NotificationRequest`, when it is type calendar, is already returning 1-12 months. So, even when yearly will ask for 0-11 months platform independent, after scheduled, it will get returned as a calendar type with 1-12 months in iOS…

Ideally, JS API should use 0-11 months entirely, in both to and from native, but that would be a breaking change for calendar type in iOS.

P.S.: I noticed a "focused" test slipped in my initial commit, sorry about that 😳  … it is now running everything

Reply:
Couldn't we intercept the `month` response from iOS in `EXNotificationSerializer.m` in `serializedDateComponents` method, and manually `- 1 ` from the month before serializing to the user?

Reply:
I guess we could, but as mentioned in the previous comment, I think any change transforming what's coming from iOS would be a breaking change, because all these Daily/Weekly/Yearly triggers are just facades for the the CalendarTrigger in iOS's case, which just plainly communicates the values *to* and *from* iOS directly. So, when we create a Daily/Weekly/Yearly trigger in iOS, and then get the scheduled triggers from iOS, we will get `UNCalendarNotificationTrigger` types in all three cases, which is the same one you get when you create a CalendarTrigger (available only in iOS). So if we change the return value in `UNCalendarNotificationTrigger` we would break existing CalendarTrigger, which are currently working with 1-12 based months. I don't like that, it's an inconsistency, but fixing it would be a breaking change I think.

Reply:
Maybe there should be another change later that standardizes everything, so in all JS we only speak 0-11 months, and in that case we could change the `EXNotificationSerializer.m` so it returns a UNCalendarNotificationTrigger type with 0-11 months, which would also fix Yearly output… but that would be a breaking change, as discussed

Reply:
> Maybe there should be another change later that standardizes everything, so in all JS we only speak 0-11 months, and in that case we could change the EXNotificationSerializer.m so it returns a UNCalendarNotificationTrigger type with 0-11 months, which would also fix Yearly output… but that would be a breaking change, as discussed

@raulmt could you open a new issue for this? I'll assign it to myself and address that in a future PR 😄